contracts/Proposal.sol

```diff
@@ -0,0 +1,120 @@
+pragma solidity ^0.5.8;
+
+import "./zeppelin/token/ERC20/IERC20.sol";
+import "./zeppelin/token/ERC20/SafeERC20.sol";
+import "./ownership/Ownable.sol";
+import "./Basket.sol";
+
+/**
+ * A Proposal represents a suggestion to change the backing for RSV.
+ *
+ * The lifecycle of a proposal:
+ * 1. Creation
+ * 2. Acceptance
+ * 3. Completion
+ *
+ * A time can be set during acceptance to determine when completion is eligible.
+ * A proposal can also be closed before it is Completed. 
+ *
+ * This contract is intended to be used in one of two possible ways. Either:
```

Comment:
These are different enough types of proposals that maybe we should separate them out. `BasketProposal` and `ExchangeProposal` with a separate `Proposal` they can both extend for the overlap between them?

Reply:
Yep, nice idea. That's the design Matt landed on too.

Reply:
Yep. This is true in `thursday`.